library/alloc/Cargo.toml

```diff
@@ -38,3 +38,12 @@ compiler-builtins-mangled-names = ["compiler_builtins/mangled-names"]
 compiler-builtins-weak-intrinsics = ["compiler_builtins/weak-intrinsics"]
 # Make panics and failed asserts immediately abort without formatting any message
 panic_immediate_abort = []
+
+[lints.rust.unexpected_cfgs]
+level = "warn"
+check-cfg = [
```

Comment:
Could you add a comment saying that these `check-cfg` entries do not yet take effect for rust-lang/rust since we use beta Cargo, but that for users for -Zbuild-std it does take effect and that the unused warning in waiting for https://github.com/rust-lang/cargo/pull/13925 to reach beta.